xl/playlist.py

```diff
@@ -294,7 +294,8 @@ def get_track_export_path(self, playlist_path, track_path, options):
         if options is not None and options.relative:
             playlist_file = Gio.File.new_for_uri(playlist_path)
             # Strip playlist filename from export path
-            export_path = playlist_file.get_uri()[:-len(playlist_file.get_basename())]
```

Comment:
That is a much more elegant way of addressing this issue. Testing this change shows that it also works. I will make a new commit for this.